### ISSUE TITLE
Reference to other Blocks

### ISSUE THREAD

**Issue by davidsger** [toeverything/AFFiNE#6087]
### Description

Would be great to have the possiblility to reference to other parts on the page like headlines or bulletpoints.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by EYHN** [comment]
We have implemented this to some extent and will bring this feature to you in 0.14.0 (Mid-April)

**Comment by EYHN** [comment]
Since we won't be tracking the development details of this feature here, I'll close this issue.

**Comment by EYHN** [comment]
Discussion in this issue:
https://github.com/toeverything/AFFiNE/issues/5907

**Comment by davidsger** [comment]
Sorry but that's not the same. I meant the reference to other parts on the page like headings or bulletpoints. Something like referencing an anchor. In Markdown it is done with \[Name\](#anchor).

**Comment by EYHN** [comment]
I think, they are implemented using the same method in affine. The synced block can be embedded or displayed as a link.

**Comment by davidsger** [comment]
Okay, I think I will see it in one of the next versions. Thanks.

**Comment by Danie10** [comment]
This needs to be re-opened as it was not actually implemented as intended. The @ search only shows other documents, and no headings or other blocks (tested in Nov 2025 on v0.25.1, even with the Indexer service running).

It's true that the original Markdown standard did not include this, but the CommonMark+ standard did, and quite a few other editors have been long supporting it. This means, too, that when importing those Markdown files, AFFiNE does not recognise the links, and they all break. The common format used is `[Go to section](#heading)`.

Yes, we can right now create a link from the target heading, and then go back to where we want it, and paste the link in there. That does work, but it breaks compatibility with many other editors, and is a bit clunky with long documents.

In summary:
1. Need the search palette to also show existing headings (ideally also tables, embedded media) for the existing document.
2. Needs to recognise and render the common format `[Go to section](#heading)` for interoperability with other apps.
